classes/publication/DAO.php

```diff
@@ -370,10 +348,10 @@ protected function deleteAuthors(int $publicationId)
      */
     protected function setControlledVocab(Publication $publication)
     {
-        $publication->setData('keywords', $this->submissionKeywordDao->getKeywords($publication->getId()));
-        $publication->setData('subjects', $this->submissionSubjectDao->getSubjects($publication->getId()));
-        $publication->setData('disciplines', $this->submissionDisciplineDao->getDisciplines($publication->getId()));
-        $publication->setData('supportingAgencies', $this->submissionAgencyDao->getAgencies($publication->getId()));
+        $publication->setData('keywords', SubmissionKeywordVocab::getKeywords($publication->getId()));
```

Comment:
Something to work on: we're trying to avoid relying on static methods, as they're hard to mock or extend. This might be better placed in a Repository class.